src/core/mac/mac.cpp

```diff
@@ -1060,7 +1136,23 @@ void Mac::BeginTransmit(void)
             sendFrame.SetSequence(mDataSequence++);
         }
         break;
-#endif
+
+#if OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
```

Comment:
Add
```c++
#if OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
    sendFrame.SetTxPeriod(0);
#endif
```
at the start of `Mac::BeginTransmit()` to avoid the frame following a CSL transmission being scheduled according to CSL sample window in `SubMac`.